proxy_test.go

```diff
@@ -310,6 +310,16 @@ func (suite *ProxyTests) TestServeHTTP_allowed() {
 	proxy.ServeHTTP(w, r)
 	assert.Equal(suite.T(), 200, w.Result().StatusCode)
 	assert.Equal(suite.T(), true, suite.fakeServer.PingedAndRestore())
+
+	// Filenames with platform incompatible characters are disallowed
+	// not checked in TestServeHTTP_allowed() because we look for a non 403 response
+	r.Method = "PUT"
+	r.URL, _ = url.Parse("/username/fi|le")
+	w = httptest.NewRecorder()
+	proxy.ServeHTTP(w, r)
+	assert.Equal(suite.T(), 406, w.Result().StatusCode)
```

Comment:
you could use `http.StatusNotAcceptable` instead of `406` here as well for readability, but might be a case for testing with the integer code as well.